### ISSUE TITLE
Storing location data in metadata tags in the clips

### ISSUE THREAD

**Issue by alexanderadam** [KyleKun/one_second_diary#100]
Hi buddy, it's me again :laughing: 

It would be nice to fill the location metadata if the location checkbox was checked.

At least these two

```yaml
TAG:location=+25.0731+121.3663/
TAG:location-eng=+25.0731+121.3663/São Paulo
```

See also [this SO thread](https://stackoverflow.com/questions/65231616/what-is-the-difference-between-location-and-location-eng-metadata-of-a-mp4-f) and [the specification](https://www.etsi.org/deliver/etsi_ts/126200_126299/126244/09.02.00_60/ts_126244v090200p.pdf).

Bonus points if not only the ISO and formatting plus name would be stored, but also the localisation that the Android phone has configured.
I think that the ISO 639-2 and 639-3 codes are used here.

So for Portuguese this would be additionally an entry called

```yaml
TAG:location-por=+25.0731+121.3663/São Paulo
```

Various tools read this metadata and it is currently gathered anyway if the checkbox is activated.

Also a map could be integrated later on which would be pretty cool.

What do you think?

**Comment by KyleKun** [comment]
I need to make a quick fix in the experimental file picker added in v1.5.2 so might as well implement this together later today, we already have the values as you said. I just don't think `location-xyz` in necessary, `location` also supports the name after the coordinates. ChatGPT seems to agree with me lol. What do you think?

Great idea on having a map, please open a separate issue later with some more info on how you imagine it looks like / what info to display - but for now let's populate that metadata since it's a prerequisite for that. Thanks!!

**Comment by alexanderadam** [comment]
For `location` and `location-eng` this is likely the same but many places have different names in their respective language in comparison to English (i.e. `Brasil` vs `Brazil` vs `Brasilien`).

It's just a luxury feature. :wink:

But `location` and `location-eng` should indeed be the same if I understand this correctly.
So I guess it wouldn't harm to have at least these two even if it's doubled. They are just very tiny text fields after all.

Tools could simply catch up and read what they want.

**Comment by KyleKun** [comment]
Makes sense, in that case maybe it's better then to store only the coordinates on `location` and the localized name in `location-xyz` based on the app language as you first proposed:

```
TAG:location=+25.0731+121.3663/
TAG:location-eng=+25.0731+121.3663/São Paulo
```

So any text won't influence other tools that might take it into account and we can decode the name anyway with the coordinates based on app language for OSD, do you agree?

As for general metadata, not much is stored today:
`artist` = app name and version
`album` = profile selected
`comment=origin` = gallery or in-app recording

I did those so any potential issues could debugged by knowing which video was made with which settings, but agree with you that much more can be stored for the sake of better filtering later on. Would appreaciate any proposals of a full list of metadata we could store, so feel free to open another issue 😅

**Comment by alexanderadam** [comment]
> Makes sense, in that case maybe it's better then to store only the coordinates on `location` and the localized name in `location-xyz` based on the app language as you first proposed:
> 
> ```yaml
> TAG:location=+25.0731+121.3663/
> TAG:location-por=+25.0731+121.3663/São Paulo
> ```
> 
> So any text won't influence other tools that might take it into account and we can decode the name anyway with the coordinates based on app language for OSD, do you agree?

Since I saw [your other nice idea](https://github.com/KyleKun/one_second_diary/issues/79) with storing custom locations (i.e. `Parent's house`) I would rather propose something like

```yaml
TAG:location=+25.0731+121.3663/São Paulo
TAG:location-por=+25.0731+121.3663/Parent's house
```

This way `location` is always the standardized thing it should be and the localized thing contains what the user would like to see. Thus keeping both information in the file.

> feel free to open another issue 😅

:laughing: 
By the way, it's really nice how OSD developed. Much happened and in my bubble some folks started using it now.

**Comment by KyleKun** [comment]
I'm experimenting with it and for some reason ffmpeg seems to ignore any text after `/`. I am trying to save `-metadata location-eng="+37.4226711-122.0849872/Mountain View, United States"` but only the coordinates are shown both in ffmpeg and ffprobe, I wonder if this is ffmpeg version based or something else. So what if we do this:
- `location` = coordinates only
- `grouping` or `comment=place` = same text that is rendered on the video, auto localized or can be edited anyway with the set custom locale

> By the way, it's really nice how OSD developed. Much happened and in my bubble some folks started using it now.

This makes me happy really, as always thanks for all the feedback 🤘

**Comment by alexanderadam** [comment]
> I am trying to save `-metadata location-eng="+37.4226711-122.0849872/Mountain View, United States"` but only the coordinates are shown both in ffmpeg and ffprobe, I wonder if this is ffmpeg version based or something else. So what if we do this:
> 
>     * `location` = coordinates only
> 
>     * `grouping` or `comment=place` = same text that is rendered on the video, auto localized or can be edited anyway with the set custom locale

Well, if there's no error, you can keep the name when storing in the `location` part. Maybe that's indeed a bug in `ffmpeg`.
Unfortunately [they're still using `trac` as an issue tracker](https://trac.ffmpeg.org) and [the search](https://trac.ffmpeg.org/search?q=location+metadata&noquickjump=1&milestone=on&ticket=on&wiki=on) is not very helpful. :grimacing: 

But yeah, we probably need to store it in another field in the meantime. `comment` or `grouping` could work. Although the proper `location` field would be best.

Did you also try to store the name in `location-eng`? Maybe only the localized variants allow to store names?

**Comment by KyleKun** [comment]
> Did you also try to store the name in `location-eng`? Maybe only the localized variants allow to store names?

Yeah, after some tests it's clear that `location` works with text afterwards and somehow seems to rule any other localized version, I can't seem to edit `location-xyz` without also editing `location`. So while there's no input for tags yet that would allow the example `Parent's house`, it can be set in as a custom locale anyway while not picking the coordinates from gps. We could simply allow to select a place in the map to edit these coordinates in the future and then it would appear correctly in the map.

In short, I will implement `location` only, and it will store coordinates (0 if gps not enabled) + name (localized).

**Comment by alexanderadam** [comment]
You're amazing! That's super cool!
